icaruscode/Generators/HepMCFileGen_module.cc

```diff
@@ -0,0 +1,457 @@
+ *  0 1
+ *  1 13 0 0 0 0 0. 0. 1.0 5.0011 0.105 1.0 1.0 1.0 0.0
+ *  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+ *  Or if you want to simulate a muon neutrino event (see oneneutrino.hepmc): 
+ *  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+ *  0 3
+ *  0 14 0 0 0 0 0.00350383 0.002469 0.589751 0.589766 0 208.939 63.9671 10.9272 4026.32
+ *  1 13 1 0 0 0 -0.168856 -0.0498011 0.44465 0.489765 105.658 208.939 63.9671 10.9272 4026.32
+ *  1 2212 1 0 0 0 0.151902 -0.124578 0.0497377 0.959907 938.272 208.939 63.9671 10.9272 4026.32
+ *  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+ *  here the first particle is the initial state neutrino (status code 0, meaning initial state, 
+ *  not to be prooagated by GEANT). The other particles are the initial state particles.
+ *  For the neutrino, write ccnc in place of 1st daugther, and mode (qe, res, ...) 
+ *  in place of 2nd daugther.
+ *
+ *  There are some assumptions that go into using this format that may not
+ *  be obvious.  The first is that only particles with status code = 1
+ *  are tracked in the LArSoft/Geant4 combination making the mother daughter
+ *  relations somewhat irrelevant. That also means that you should let
+ *  Geant4 handle any decays.
+ *
+ *  The units in LArSoft are cm for distances and ns for time.
+ *  The use of `TLorentzVector` below does not imply space and time have the same units
+ *   (do not use `TLorentzVector::Boost()`).
+ */
+#include <string>
+#include <fstream>
+#include <math.h>
+#include <map>
+#include <iomanip>
+#include <algorithm>
+#include <sstream>
+#include <glob.h>
+#include <cstdlib>  // for unsetenv()
+#include "art/Framework/Core/EDProducer.h"
+#include "art/Framework/Core/ModuleMacros.h"
+#include "art/Framework/Principal/Event.h"
+#include "art/Framework/Principal/SubRun.h"
+#include "fhiclcpp/ParameterSet.h"
+#include "cetlib_except/exception.h"
+#include "art/Framework/Services/Registry/ServiceHandle.h"
+#include "cetlib/search_path.h"
+#include "cetlib/getenv.h"
+#include "cetlib/split_path.h"
+#include "cetlib_except/exception.h"
+#include "messagefacility/MessageLogger/MessageLogger.h"
+#include "TLorentzVector.h"
+#include "larcore/Geometry/Geometry.h"
+#include "larcoreobj/SummaryData/RunData.h"
+#include "larcoreobj/SummaryData/POTSummary.h"
+#include "nusimdata/SimulationBase/MCTruth.h"
+#include "nusimdata/SimulationBase/MCParticle.h"
+#include "ifdh.h"       // use bare
+#undef USE_IFDH_SERVICE // ifdh for now
+// #ifndef NO_IFDH_LIB
+//   #define USE_IFDH_SERVICE 1
+//   // IFDHC
+//   #ifdef USE_IFDH_SERVICE
+//     #include "IFDH_service.h"
+//   #else
+//     // bare IFDHC
+//     #include "ifdh.h"
+//   #endif
+// #else
+//   #undef USE_IFDH_SERVICE
+//   // nothing doing ... use ifdef to hide any reference that might need header
+//   #include <cassert>
+// #endif
```

Comment:
Not clear to me if this needs to stay. It smells outdated, although I am not 100% sure.
```suggestion
```
(right _now_ this module does not use IFDH at all, but that's a slightly different point)